### PR TITLE
MINOR: Remove deprecated `host.name`, `advertised.host.name`, `port` and `advertised.port` for 3.0

### PR DESCRIPTION
**Pull request by dajac**
`host.name`, `port`, `advertised.host.name` and `advertised.port` were deprecated a long time ago in favour of using `listeners`.  I think that it is time to remove them.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Makes sense. I assume you're planning to file a JIRA and the upgrade note still?

**Comment by dajac** [comment]
@ijuma Yes, will do. The PR is not fully completed yet. I'll ping you once it is ready.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -1804,9 +1773,7 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
   // If the user did not define listeners but did define host or port, let's use them in backward compatible way
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:1773]: Maybe you saw this already, but this comment needs to be updated.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 2]
```diff
@@ -1838,8 +1805,6 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
     val advertisedListenersProp = getString(KafkaConfig.AdvertisedListenersProp)
     if (advertisedListenersProp != null)
       CoreUtils.listenerListToEndPoints(advertisedListenersProp, listenerSecurityProtocolMap, requireDistinctPorts=false)
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:1807]: the docs of this method need to be updated.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/KafkaEmbedded.java [hunk 3]
```diff
@@ -108,8 +107,10 @@ private Properties effectiveConfigFrom(final Properties initialConfig) {
     @SuppressWarnings("WeakerAccess")
     public String brokerList() {
         final Object listenerConfig = effectiveConfig.get(KafkaConfig$.MODULE$.InterBrokerListenerNameProp());
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/KafkaEmbedded.java:109]: not sure why it tried to get inter listeners. The return value of this method is used for producer/consumer.

File: connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java [hunk 4]
```diff
@@ -232,7 +232,9 @@ public String bootstrapServers() {
     }
 
     public String address(KafkaServer server) {
-        return server.config().hostName() + ":" + server.boundPort(listenerName);
+        final String hostName = JavaConverters.asJavaCollection(server.config().advertisedListeners())
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java:235]: Maybe we can move this method to `KafkaConfig`. It is useful for testing (connect and streams)

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 5]
```diff
@@ -101,7 +101,7 @@ public void start() throws IOException {
 
         for (int i = 0; i < brokers.length; i++) {
             brokerConfig.put(KafkaConfig$.MODULE$.BrokerIdProp(), i);
-            log.debug("Starting a Kafka instance on port {} ...", brokerConfig.get(KafkaConfig$.MODULE$.PortProp()));
+            log.debug("Starting a Kafka instance on {} ...", brokerConfig.get(KafkaConfig$.MODULE$.ListenersProp()));
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java:104]: The default port is 0 so the return value of `brokerConfig.get(KafkaConfig$.MODULE$.ListenersProp())` is 0, right? If so, it is weird to log it.

**Review by chia7712 — COMMENTED** [review]
@dajac nice cleanup!

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (showuon) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:1773]: +1

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 2 above)
Comment (showuon) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:1807]: +1

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 6]
```diff
@@ -143,73 +143,6 @@ class KafkaConfigTest {
...
-    props.put(KafkaConfig.PortProp, port)
-    val serverConfig = KafkaConfig.fromProps(props)
-    val endpoints = serverConfig.advertisedListeners
-    val endpoint = endpoints.find(_.securityProtocol == SecurityProtocol.PLAINTEXT).get
-    assertEquals(endpoint.host, hostName)
-    assertEquals(endpoint.port, port.toInt)
-  }
-
-  @Test
-  def testAdvertiseConfigured(): Unit = {
```
Comment (showuon) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala:163]: I saw you removed all the `AdvertiseListener`'s tests here. Should we add or keep and update some tests for it?

**Review by showuon — COMMENTED** [review]
@dajac , thanks for the PR. Left some comments. Thanks.

**Comment by showuon** [comment]
Also, the jenkins build didn't complete last time. Please remember to trigger it again. Thanks.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java (hunk 5 above)
Comment (ijuma) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java:104]: This is logging the configured listener and after the broker is started, the resolved listener is logged. The default could have been changed, so this seems ok. Right?

**Comment by dajac** [comment]
Replaced by https://github.com/apache/kafka/pull/10872.

**Pull request closed without merging**